src/main/java/org/candlepin/subscriptions/db/EventRecordRepository.java

```diff
@@ -72,4 +72,10 @@
      */
     Stream<EventRecord> findByAccountNumberAndEventSourceAndEventTypeAndTimestampGreaterThanEqualAndTimestampLessThanOrderByTimestamp(
         String accountNumber, String eventSource, String eventType, OffsetDateTime begin, OffsetDateTime end);
+
+    /**
+     * Delete old event records given a cutoff date
+     * @param cutoffDate Dates BEFORE this timestamp get deleted
+     */
+    void deleteEventRecordsByTimestampBefore(OffsetDateTime cutoffDate);
```

Comment:
This is fairly minor, but is it possible to return the number of records deleted?  I was thinking it'd be nice to have in the log messages "Blah blah job deleted N records" so that you can tell if something actually happened.

Reply:
Agreed, I actually had something like this at first.  I can't do it using spring data "out of the box", but you could do something where you do the delete in two parts:

```java
var results = repo.findAllByTimestampBefore(timestamp);
int numberToDelete = results.size();

repo.delete(results);

log.info("deleted {} event records", numberToDelete);
```

We could technically explicitly define the Query to get the number back too, but I've been trying to avoid explicitly defining queries using the annotation like this - so we can use spring data to enhance them later if we need to.
```java
@Modifying
@Query("delete from EventRecords where timestamp Before :cutoffDate")
public int deleteByTimestampBefore(OffsetDateTime cutoffDate);
```

I'm interested in everyones' opinions on how we want to handle deletes like this.

@awood @kahowell  @mstead @dmckoske

Reply:
IMO, if Spring Data doesn't do it out of the box, then I don't think it's important enough to implement custom functionality.